src/vue-components/modals/messageBox.vue

```diff
@@ -0,0 +1,262 @@
+<template>
+    <div class="modal">
+        <div class="modal-mask">
+            <div class="modal-wrapper">
+                <div class="modal-container">
+                    <a v-if="showCloseButton" id="closeLink" :title="$t('close')" class="modal-close-link black" href="javascript:;" @click="handleClose">
+                        <i class="fas fa-times"></i>
+                    </a>
+
+                    <div class="modal-header">
+                        <i :class="iconClass"></i>
+                        <h1>{{header}}</h1>
+                    </div>
+
+                    <div class="modal-body">
+                        <div class="message-text">{{message}}</div>
+                        <ul v-if="items && items.length > 0" class="message-items">
+                            <li v-for="item in items" :key="item">{{item}}</li>
+                        </ul>
+                    </div>
+
+                    <div class="modal-footer" v-if="buttons && buttons.length > 0">
+                        <button
+                            v-for="(button, index) in buttons"
+                            :key="index"
+                            @click="handleButtonClick(button)"
+                        >
+                            <i v-if="button.icon" :class="button.icon"></i> <span>{{button.label | translate}}</span>
+                        </button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+    import './../../css/modal.css';
+    import { i18nService } from '../../js/service/i18nService';
+
+    // Modal types
+    const MODAL_TYPE_SUCCESS = 'success';
```

Comment:
all these constants could/should be moved to `constants.js` so that when opening the message box and passing the `buttonPreset` property we can also use these constants and don't have to pass magic strings.